vpr/src/pack/cluster.cpp

```diff
@@ -2566,9 +2593,30 @@ void add_cluster_molecule_candidates_by_transitive_connectivity(t_pb* cur_pb,
     }
 }
 
+bool check_free_primitives_for_molecule_atoms(t_pack_molecule* molecule, t_cluster_placement_stats* cluster_placement_stats_ptr) {
+    auto& atom_ctx = g_vpr_ctx.atom();
+    bool success = true;
+
+    for (int i_atom = 0; i_atom < get_array_size_of_molecule(molecule); i_atom++) {
+        if (molecule->atom_block_ids[i_atom]) {
+            VTR_ASSERT(atom_ctx.lookup.atom_clb(molecule->atom_block_ids[i_atom]) == ClusterBlockId::INVALID());
+            auto blk_id2 = molecule->atom_block_ids[i_atom];
+            if (!exists_free_primitive_for_atom_block(cluster_placement_stats_ptr, blk_id2)) {
+                /* TODO: debating whether to check if placement exists for molecule
```

Comment:
add Jason Luu's name so people would know who to ask about this

Reply:
This is marked as resolved but I don't see Jason Luu's name added?

Reply:
It is added at the end of the comment, I'll move it to the beginning though so it's more clear